### PR TITLE
User story 2

### PR DESCRIPTION
**Pull request by martinhaiss**
This resolves [#2](https://github.com/martinhaiss/capstone-project/issues/2)

- Add button to card
- Add function to expand or collapse activity-card when button is clicked
- Change card layout from flexbox to grid
- Add test for new feature to already existing card-tests

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,6 +12,12 @@ function App() {
 const AppLayout = styled.div`
   max-width: 600px;
   height: 98vh;
+  margin: auto;
+  display: grid;
+  grid-template-columns: 1fr;
+  grid-template-rows: 0.125fr 1fr 0.075fr;
+  grid-column-gap: 0px;
+  grid-row-gap: 5px;
```
Comment (pixxelschubser) [inline src/App.js:20]: This can be one line with the shorthand `gap` property:

```suggestion
  gap: 5px 0;
```

File: src/components/Card.js [hunk 2]
```diff
@@ -1,26 +1,62 @@
...
+  duration,
+  postalcode,
+  city,
+  street,
+  info,
+}) {
+  const [showDetails, setshowDetails] = useState(false);
+
+  const handleDetailsButtonClick = () => {
+    setshowDetails(showDetails ? false : true);
```
Comment (pixxelschubser) [inline src/components/Card.js:20]: Would the following code would do the same?

```suggestion
  const handleDetailsButtonClick = () => {
    setshowDetails(!showDetails);
```

File: src/components/Card.js [hunk 3]
```diff
@@ -31,51 +67,114 @@ const Wrapper = styled.div`
   padding: 15px;
   border-radius: 13px;
   box-shadow: 0px 0px 20px #dedede;
+  display: flex;
+  flex-direction: column;
+`;
+
+const Heading = styled.div`
```
Comment (pixxelschubser) [inline src/components/Card.js:74]: You ***could*** convert this `div`into a `header`so each card has its own actual header. That's totally optionally though.

File: src/components/Card.js [hunk 4]
```diff
@@ -1,26 +1,62 @@
...
+      {showDetails ? (
+        <div>
+          <Adress>
+            <Meet>Meeting Point</Meet>
+            <Street>{street}</Street>
+            <City>
+              {postalcode} {city}
+            </City>
+            <Info>{info}</Info>
+          </Adress>
```
Comment (pixxelschubser) [inline src/components/Card.js:52]: Minor typo: "Address" is written with two "d".

File: src/components/Card.js [hunk 5]
```diff
@@ -31,51 +67,114 @@ const Wrapper = styled.div`
...
+
+const Distance = styled.p`
+  grid-column: 2 / 4;
+`;
+
+const Duration = styled.p`
+  grid-column: 5 / 7;
+`;
+
+const Adress = styled.div`
```
Comment (pixxelschubser) [inline src/components/Card.js:125]: same as above: "Address"

File: src/components/Card.spec.js [hunk 6]
```diff
@@ -18,3 +18,10 @@ describe("Card", () => {
     expect(img).toBeInTheDocument();
   });
 });
+
+it("renders one buttons", () => {
```
Comment (pixxelschubser) [inline src/components/Card.spec.js:22]: renders one button*

File: src/components/Card.js (hunk 2 above)
Comment (macfrei) [inline src/components/Card.js:20]: I would use the `!` (NOT) operator. 
```suggestion
    setshowDetails(!showDetails);
```

File: src/components/Card.js [hunk 7]
```diff
@@ -31,51 +67,114 @@ const Wrapper = styled.div`
...
 `;
 
-const CardTitle = styled.div`
-  display: flex;
-  justify-content: flex-start;
-`;
-
-const Spacer = styled.div`
-  width: 10px;
+const Title = styled.p`
```
Comment (macfrei) [inline src/components/Card.js:87]: I would expect `Title` to be a heading element! Maybe a `h2`?

File: src/components/Card.js (hunk 2 above)
Comment (martinhaiss) [reply src/components/Card.js:20]: Already done, thx!

File: src/components/Card.js (hunk 7 above)
Comment (martinhaiss) [reply src/components/Card.js:87]: Understood, I changed it to "Name".

**Pull request merged**